sdk/swagger.yaml

```diff
@@ -0,0 +1,720 @@
+    description: Pulumi Cloud Production Preview API
+components:
+  securitySchemes:
+    Authorization:
+      type: apiKey
+      name: Authorization
+      in: header
+  schemas:
+    Error:
+      type: object
+      properties:
+        message:
+          type: string
+        code:
+          type: integer
+      required:
+        - message
+        - code
+    Range:
+      type: object
+      properties:
+        environment:
+          type: string
+        begin:
+          $ref: "#/components/schemas/Pos"
+        end:
+          $ref: "#/components/schemas/Pos"
+      required:
+      - environment
+      - begin
+      - end
+    Reference:
+      type: object
+      required:
+        - $ref
+      properties:
+        $ref:
+          type: string
+          format: uri-reference
+    Accessor:
+      type: object
+      properties:
+        index:
+          type: integer
+        key:
+          type: string
+        range:
+          $ref: "#/components/schemas/Range"
+      required:
+      - key
+      - range
+    PropertyAccessor:
+      allOf:
+        - $ref: "#/components/schemas/Accessor"
+        - type: object
+          properties:
+            value:
+              $ref: "#/components/schemas/Range"
+    Access:
+      type: object
+      properties:
+        receiver:
+          $ref: "#/components/schemas/Range"
+        accessors:
+          type: array
+          items:
+            $ref: "#/components/schemas/Accessor"
+    Interpolation:
+      type: object
+      properties:
+        text:
+          type: string
+        value:
+          type: array
+          items:
+            $ref: "#/components/schemas/PropertyAccessor"
+      required:
+      - text
+    ExprBuiltin:
+      type: object
+      properties:
+        name:
+          type: string
+        nameRange:
+          $ref: "#/components/schemas/Range"
+        argSchema: {}
+        arg:
+          $ref: "#/components/schemas/Expr"
+      required:
+      - name
+    Expr:
+      type: object
+      properties:
+        range:
+          $ref: "#/components/schemas/Range"
+        base:
+          $ref: "#/components/schemas/Expr"
+        schema: {}
+        keyRanges:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Range"
+        literal: {}
+        interpolate: 
+          type: array
+          items:
+            $ref: "#/components/schemas/Interpolation"
+        symbol:
+          type: array
+          items:
+            $ref: "#/components/schemas/PropertyAccessor"
+        access:
+          type: array
+          items:
+            $ref: "#/components/schemas/Access"
+        list:
+          type: array
+          items:
+            $ref: "#/components/schemas/Expr"
+        object:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Expr"
+        builtin:
+          $ref: "#/components/schemas/ExprBuiltin"

```

Comment:
This really ought to be a `oneOf`--only one of these properties can ever be present at a time. Do `oneOf`s play well with the various code generators?

Reply:
Sometimes, it has been hit or miss.  But a lot of this I pulled from the schema swagger doc, I will have to check if I missed a oneOf there.

Reply:
The isssue with oneOf here is that it expects that to consolidate to one property in the serialized result, like a Value property that can be oneOf several different schemas.  The generator doesn't have support how we are serializing the output here, and really doesn't have great discriminated union support in general.